File: Inkligth-main/inklight-frontend/src/api/folder.ts
import apiClient from './client'

export interface FolderItem {
  id: string
  user_id: string
  name: string
  parent_id: string | null
  literature_count: number
  created_at: string
  updated_at: string
}

export interface FolderListResponse {
  items: FolderItem[]
}

export function getFolders() {
  return apiClient.get<FolderListResponse>('/folders')
}

export function createFolder(name: string) {
  return apiClient.post<FolderItem>('/folders', { name })
}

export function renameFolder(id: string, name: string) {
  return apiClient.patch<FolderItem>(`/folders/${id}`, { name })
}

export function deleteFolder(id: string) {
  return apiClient.delete(`/folders/${id}`)
}

File: Inkligth-main/backend/app/schemas/folder.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional


class FolderCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)


class FolderUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=100)


class FolderResponse(BaseModel):
    id: str
    user_id: str
    name: str
    parent_id: Optional[str] = None
    literature_count: int = 0
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class FolderListResponse(BaseModel):
    items: list[FolderResponse]
